sound/soc/intel/boards/sof_rt5682.c

```diff
@@ -66,12 +66,29 @@ struct sof_card_private {
 	bool common_hdmi_codec_drv;
 };
 
+const char *sof_override_tplg_name;
+
 static int sof_rt5682_quirk_cb(const struct dmi_system_id *id)
 {
 	sof_rt5682_quirk = (unsigned long)id->driver_data;
 	return 1;
 }
-
+static int sof_rt5682_tplg_cb(const struct dmi_system_id *id)
+{
+	sof_override_tplg_name = (const char *)id->driver_data;
+	return 1;
+}
+static const struct dmi_system_id sof_rt5682_tplg_table[] = {
+	{
+		.callback = sof_rt5682_tplg_cb,
+		.matches = {
+			DMI_MATCH(DMI_PRODUCT_FAMILY, "Google_Volteer"),
+			DMI_MATCH(DMI_PRODUCT_NAME, "Terrador"),
+		},
+		.driver_data = (void *)("sof-tgl-max98373-rt5682-ssp2.tplg"),
```

Comment:
I think I suggested before to remove the cast:
```
.driver_data = "sof-tgl-max98373-rt5682-ssp2.tplg",
```

Reply:
sorry, I seem to have missed removing type casting. Now, I have updated it.